### ISSUE TITLE
[Question/Feature Request] Calling saga on every page change

### ISSUE THREAD

**Issue by TeoTN** [jfairbank/redux-saga-router#8]
Is there any pattern that would run some saga every time url is changed? Or could you please implement it?

**Comment by jfairbank** [comment]
Hi, @TeoTN!

I assume you mean is there a way to run a specific saga every time the route changes regardless of the specific route while still running route-specific sagas too? What would be your use case for that?

**Comment by TeoTN** [comment]
Exactly! My usecase is that I have a list of messages shown to user (notifications, errors etc), and I'd like to dispatch action `CLEAN_MESSAGES` when user navigates to another Route. (It looks weird when you have "Settings saved" after navigating to other page)

**Comment by TeoTN** [comment]
Another use case would be to check whether authentication token is still valid - some banking pages check it on changing page

**Comment by jfairbank** [comment]
This sounds like a reasonable addition to make. My only concern is how to handle situations where you _wouldn't_ want the catch-all saga to run before a particular route. I don't want to make too many assumptions, so I'd like to think through that and probably avoid initially adding it until it becomes a desired feature.

As far as adding the catch-all feature, it might take me a bit to get around to it because I'm doing some conference preparation work the next couple weeks. However, I'd happily consider a PR right now. I'm initially thinking this needs to be part of an optional third configuration object argument to the API. So something like this (not sold on the naming yet):

```js
import { router, createBrowserHistory } from 'redux-saga-router';

const history = createBrowserHistory();

const routes = {
  '/users': function* usersSaga() {
    const users = yield call(fetchUsers);
    yield put(setUsers(users));
  },

  '/users/:id': function* userSaga({ id }) {
    const user = yield call(fetchUser, id);
    yield put(setCurrentUser(user));
  },
};

const options = {
  *beforeRouteChange() {
    yield put(clearNotifications());
  },
};

function* mainSaga() {
  const data = yield call(fetchInitialData);

  yield put(ready(data));

  yield* router(history, routes, options);
}
```

**Comment by jfairbank** [comment]
Released in v2.1.0.
